### PR TITLE
Updating ShadowRealm results for Firefox

### PR DESCRIPTION
**Pull request by afmenez**
Ditto.

### CODE REVIEW

**Comment by afmenez** [comment]
Firefox 128 will be released tomorrow, it would be nice if this could be merged before that to avoid the conflicts.

File: data-esnext.js [hunk 1]
```diff
@@ -92,6 +92,12 @@ exports.tests = [
     ie11: false,
     firefox10: false,
     firefox52: false,
+    firefox103: false,
+    firefox104: {
+          val: 'flagged',
+          note_id: 'ff-shadow-realm',
+          note_html: 'The feature has to be enabled via <code>javascript.options.experimental.shadow_realms</code> setting under <code>about:config</code>.'
+        },
```
Comment (chicoxyzzy) [inline data-esnext.js]: Hmm... Probably spec has changed since then. Maybe we need to reflect that in the comment

Comment (afmenez) [reply data-esnext.js]: The last commit for this feature was on Firefox 107, I updated the patch to this version.

**Review by ljharb — APPROVED** [review]

**Comment by afmenez** [comment]
Thank you!

**Pull request merged**